File: leetcode_1207.c
bool uniqueOccurrences(int* arr, int arrSize){
    int nums[2000] = {0};
    int nums1[2000] = {0};
    for (int i = 0; i < arrSize; i++) {
        nums[arr[i] + 1000]++;
    }
    for (int i = 0; i < 2000; i++) {
        if (nums[i] && nums1[nums[i]]) {
            return false;
        }
        nums1[nums[i]] = 1;
    }
    return true;
}

File: solutions/leetcode_1207.py
from collections import Counter


class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        s = set()
        for i in Counter(arr).values():
            if i in s:
                return False
            s.add(i)
        return True


if __name__ == '__main__':
    given_arr = [1,2,2,1,1,3]
    expected = True

    solution = Solution()
    result = solution.uniqueOccurrences(given_arr)
    assert result == expected
